services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/files.py

```diff
@@ -53,3 +54,27 @@ async def delete_file(
         api_secret=x_datcore_api_secret,
         obj_id=file_id,
     )
+
+
+@router.get(
+    "/packages/{package_id}/files",
+    summary="returns a package (i.e. a file)",
+    status_code=status.HTTP_200_OK,
+    response_model=list[dict[str, Any]],
+)
+@cancel_on_disconnect
+async def get_package(
+    request: Request,
+    package_id: str,
+    x_datcore_api_key: str = Header(..., description="Datcore API Key"),
+    x_datcore_api_secret: str = Header(..., description="Datcore API Secret"),
+    pennsieve_client: PennsieveApiClient = Depends(get_pennsieve_api_client),
+) -> list[dict[str, Any]]:
+    assert request  # nosec
+    return await pennsieve_client._get_package_files(  # noqa: SLF001
```

Comment:
Generally it is not a good idea to use protected functions . We risk to break backwards compatibility at some point. Please make sure (e.g. with tests) that this entry will always be up-to-date

Reply:
I guess ```_get_package_files``` should be a public function (i.e. no _ in front)

Reply:
I can make `_get_package_files` a public function, unless originally there was a reason to have it as a private function (it was like that in the previous version of the code).